### PR TITLE
Proposal: initial project setup

### PR DESCRIPTION
**Pull request by nicolas-goudry**
This PR setup the repository with the following:

* basic git stuff
  * gitignore
  * husky: for version controlled hooks
* enforce commit style: uses commitlint with config-conventional
* enforce code style: uses standard with comma-dangle custom rule (for easier reviews)
* add Github issue templates for:
  * feature requests
  * issues
* add main readme file with some todo
* add contributing guide with code of conduct

Let me know what you think about this!

cc @brochington

### CODE REVIEW

**Comment by brochington** [comment]
All of this so far looks good to me. Thank you for doing it. :shipit: 

I would like propose using The Typescript “language” (probably not the actual tsc compiler) for this project. I’ve found it invaluable. I can set it up.

What are preferences for testing? Jest?

**Comment by nicolas-goudry** [comment]
Never used Typescript, so I could use some help, but I’m not against it, go ahead :wink: 

Regarding testing, I’m more of a backend guy, so I never used Jest, but you seems to be more front-end than me, so I’ll let you chose!

Should we wait for this to be merged, or maybe you could push your work on my fork so it’s added to this PR?

**Comment by brochington** [comment]
Copy on using Typescript. Happy to help you with it, if needed. 

Yeah, I do mostly FE work nowadays. Jest seems to be a good standard testing library, So I can add that too. 

I don't mind waiting a bit for the PR to be merged, as long as it's not going to take like...weeks. @lipis Any guidance on moving forward with things like PRs?

**Comment by nicolas-goudry** [comment]
I heard about [testing library](https://testing-library.com/) getting much traction lately, ever heard of it?

**Comment by lipis** [comment]
Hey didn't forget you.. I just didn't get the chance! Will get back to you soon.

**Comment by lipis** [comment]
This looks great few things:

- Let's start simple and not include the templates for issues/PRs
- Same for code of conduct.. we're missing one for the core, and once we'll get it we'll use it on all projects (most likely)
- For Prettier user [`@excalidraw/prettier-config`](https://www.npmjs.com/package/@excalidraw/prettier-config) just released :)
- For ESlint we want to make something similar to have the same for all projects as we grow (https://github.com/excalidraw/eslint-config), but it's still under construction :) Let's leave it for now but just a heads up.
- For commits we are using [husky](https://github.com/typicode/husky) and [lint-staged](https://github.com/okonet/lint-staged). I never seen your setup, but let's fly with it and we might change it later.

**Comment by lipis** [comment]
I also invited you @nicolas-goudry to the https://github.com/orgs/excalidraw/teams/svg-to-excalidraw.. anyone else that we need to get there?

**Comment by nicolas-goudry** [comment]
I’ll make these adjustments tonight and merge this PR.

Duly noted for ESLint config, we’ll try to keep an eye on it, I’ll just raise an issue in this repo so we don’t forget.

I’m not familiar with lint-staged, I’ll give it a look. Husky is set up. I just added commitlint to enforce commits style.

Us two are the only ones working on this for now, so it’s ok!

Thanks for everything :wink:

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+dist
+node_modules
```
Comment (lipis) [inline .eslintignore:2]: end of line

**Comment by lipis** [comment]
We also recently switched to Yarn..

**Comment by nicolas-goudry** [comment]
@lipis I think I didn’t forget anything!

- removed code of conduct
- removed Github issue templates
- switched to @excalidraw/prettier-config (and ran it)
- setup lint-staged (with husky v5)
- fixed missing EOL
- switched to yarn
- setup Github Actions workflows

**Review by lipis — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1 +1,70 @@
...
+- [:beginner: Usage][usage]
+- [:game_die: Running tests][running-tests]
+- [:building_construction: Local Development][local-development]
+- [:busts_in_silhouette: Contributing][contributing]
+- [:1234: Versioning][versioning]
+
+## :floppy_disk: Installation
+
+```bash
+npm install svg-to-excalidraw --save # or yarn add svg-to-excalidraw
```
Comment (lipis) [inline README.md:19]: yarn

File: README.md [hunk 3]
```diff
@@ -1 +1,70 @@
...
+
+### :building_construction: Local Development
+
+#### Building the Project
+
+```bash
+npm run build # builds the project
+# or
+npm run build:watch # builds project whenever a file is updated.
+```
```
Comment (lipis) [inline README.md:38]: yarn

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+const baseConfig = {
+  entry: [path.join(process.cwd(), "src/index.ts")],
+  // output: {
+  //   filename: 'bundle.js',
+  //   path: path.resolve(process.cwd(), 'dist'),
+  //   library: 'svg-to-excalidraw',
+  //   libraryTarget: 'umd',
+  //   publicPath: '/static/',
+  //   umdNamedDefine: true,
+  // },
```
Comment (lipis) [inline webpack.config.js:26]: use it or remove it..

File: package.json [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    "eslint": "^7.20.0",
+    "eslint-config-prettier": "^7.2.0",
+    "eslint-plugin-import": "^2.22.1",
+    "husky": "^5.0.9",
+    "lint-staged": "^10.5.4",
+    "prettier": "^2.2.1",
+    "serve": "^11.3.2",
+    "typescript": "^4.1.5",
+    "webpack": "^5.23.0",
+    "webpack-cli": "^4.5.0"
```
Comment (lipis) [inline package.json:64]: pinned versions for less surprises..

**Pull request merged**